crates/usvg/src/writer.rs

```diff
@@ -668,12 +709,98 @@ fn conv_element(node: &Node, is_clip_path: bool, opt: &XmlOptions, xml: &mut Xml
                 );
             }
 
-            conv_elements(node, false, opt, xml);
+            conv_elements(node, false, writer_context, xml);
 
             xml.end_element();
         }
-        NodeKind::Text(_) => {
-            log::warn!("Text must be converted into paths.");
+        NodeKind::Text(ref text) => {
+            xml.start_svg_element(EId::Text);
+
+            if !text.id.is_empty() {
+                xml.write_id_attribute(&text.id, writer_context.opt);
+            }
+
+            xml.write_attribute_raw("xml:space", |buf| {
```

Comment:
Isn't this just `xml.write_attribute_raw("xml:space",  "preserve")`?

PS: yes, you got it right, usvg text should always be marked as `preserve` and no new lines must be written.